src/wallet/wallet.cpp

```diff
@@ -2480,6 +2480,17 @@ void CWallet::UpdatedTransaction(const uint256 &hashTx)
     }
 }
 
+void CWallet::GetScriptForMining(CScript &script)
+{
+    CReserveKey reservekey(this);
+    reservekey.KeepKey();
```

Comment:
As the internal miner is meant for testing only, this simple solution is elegant. I'd prefer not to introduce complex micro-management of keys, unless it's necessary for anything besides the miner. Responsibility for not requesting absurd numbers of scripts can be put with the callers (e.g. the mining threads).